validator/src/main.rs

```diff
@@ -370,6 +370,22 @@ fn hardforks_of(matches: &ArgMatches<'_>, name: &str) -> Option<Vec<Slot>> {
     }
 }
 
+fn get_observable_vote_accounts(matches: &ArgMatches<'_>) -> HashSet<Pubkey> {
+    let vote_account = if matches.is_present("vote_account") {
+        vec![pubkey_of(&matches, "vote_account").unwrap()]
```

Comment:
I suppose things fail elsewhere already if you pass something invalid to `--vote-account`? But to make it a bit safer, instead of unwrapping, we could preserve the `Option` (or convert it into one with `.ok()` if it was a `Result`), and then in the `else` arm return `None`. `Option` implements `IntoIterator` so it can be used with `.extend` below.

Or even just move the check down, and insert it conditionally.

**Edit**: Ah, the unwrap is safe because Clap already validates the value ... In that case I prefer to use `.expect()` and put the reason for why it is safe in there. (It also helps to keep superficial auditors at bay who say “unwrap is bad, mkay?”.)